src/components/UnitModal/UnitModal.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { FC } from "react";
+
+import Flex from "@/components/Flex";
+import Box from "@/components/Box";
+import { Heading } from "@/components/Typography";
+import Button from "@/components/Button";
+import { Unit } from "@/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab";
+import { TagFunctional } from "@/components/TagFunctional";
+import LessonMetadata from "@/components/LessonMetadata";
+import BrushBorders from "@/components/SpriteSheet/BrushSvgs/BrushBorders";
+import Card from "@/components/Card";
+
+type UnitModalProps = {
+  unitData: Unit | null;
+};
+
+const UnitModal: FC<UnitModalProps> = ({ unitData }) => {
+  if (!unitData) return null;
```

Comment:
Should this be in the JSX return statement? like a `unitData ? <><> : null` It might be overlooked up at the top of the component

Reply:
So this is a type guard for the unitOptionsAvailable boolean being constructed, not the rendering of the component (weird I know)  it's being removed on my next PR